compiler/parser/parser.go

```diff
@@ -1016,15 +1016,27 @@ func (p *Parser) parseMapElementLit() *ast.MapElementLit {
 	}
 
 	// key: read identifier token but it's not actually an identifier
-	ident := p.parseIdent()
+	pos := p.pos
+	name := "_"
+
+	if p.token == token.Ident {
+		name = p.tokenLit
+		p.next()
+	} else if p.token == token.String {
+		v, _ := strconv.Unquote(p.tokenLit)
+		name = v
+		p.next()
+	} else {
+		p.expect(token.Ident)
```

Comment:
What about this?
```golang
p.errorExpected(pos, "map key") // or maybe "string"?
p.next()
```

Then, you might want to move `p.next()` to below `if-else if-else` statement.

Reply:
Yeah, that looks better. "map key" is perhaps more suitable then "string" as both token.String and token.Ident are allowed.